src/partition/repart.c

```diff
@@ -2803,6 +2803,9 @@ static int context_discard_range(
         assert(offset != UINT64_MAX);
         assert(size != UINT64_MAX);
 
+        if (!arg_discard)
+                return 0;
+
```

Comment:
we usually do this test earlier, for the other case, to skip the whole looping and stuff. 

I think to match that logic we should probably also do the same for the issue at hand here, and add the test to context_write_partition_table() right before the context_discard_range() call, which is the only call of that function that isn#t already covered by arg_discard

Reply:
updated